src/plugins/index_pattern_management/public/components/field_editor/field_editor.tsx

```diff
@@ -411,6 +414,33 @@ export class FieldEditor extends PureComponent<FieldEdiorProps, FieldEditorState
     );
   }
 
+  renderCustomName() {
+    const { customName, spec } = this.state;
+
+    return (
+      <EuiFormRow
+        label={i18n.translate('indexPatternManagement.customNameLabel', {
+          defaultMessage: 'Custom name',
+        })}
+        helpText={
+          <FormattedMessage
+            id="indexPatternManagement.customNameHelpText"
+            defaultMessage="Set a custom name to use when this field is displayed in Discover and Lens. Queries and filters don't currently support a custom name and will use the original field name."
```

Comment:
nit: This mentions Lens specifically, but in the meantime Visualize supports custom names as well - we might want to adjust the wording

Reply:
thx I've changed it!